### PR TITLE
Volumetric height fog

### PR DESCRIPTION
**Pull request by EvgeniiG**
### Purpose of this PR
Adds the initial implementation of volumetric height fog

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl [hunk 1]
```diff
@@ -55,7 +55,16 @@ void EvaluateLight_Directional(LightLoopContext lightLoopContext, PositionInputs
     float  shadowMask = 1.0;
 
     color       = light.color;
-    attenuation = 1.0; // TODO: implement volumetric attenuation along shadow rays for directional lights
+    attenuation = 1.0;
+
+    // Height fog attenuation.
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: should we test that fog (or volumetric fog?) is enabled at this point?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl [hunk 2]
```diff
@@ -55,7 +55,16 @@ void EvaluateLight_Directional(LightLoopContext lightLoopContext, PositionInputs
...
+    {
+        float cosZenithAngle = L.z;
+        float fragmentHeight = GetAbsolutePositionWS(posInput.positionWS).y;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: You should convert the fog to be camera relative and not convert the position to absolute here.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl (hunk 1 above)
Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: For consistent visuals we must volumetrically shadow all surfaces and volumes the same way.

Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: If we want to claim our lighting is "unified".

Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: I agree that there's extra cost, though.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl (hunk 2 above)
Comment (EvgeniiG) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: Done.

**Pull request merged**